nncf/common/tensor_statistics/statistic_point.py

```diff
@@ -29,19 +29,19 @@ def __init__(self, target_point: TargetPoint, tensor_collector: TensorStatisticC
         self.target_point = target_point
         self.algorithm_to_tensor_collectors = {algorithm: [tensor_collector]}
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> Any:
         return (
             self.target_point == other.target_point
             and self.algorithm_to_tensor_collectors == other.self.algorithm_to_tensor_collectors
         )
 
-    def register_tensor(self, x: TensorType):
+    def register_tensor(self, x: NNCFTensor) -> None:
         for tensor_collectors in self.algorithm_to_tensor_collectors.values():
             for tensor_collector in tensor_collectors:
                 tensor_collector.register_input(x)
 
 
-class StatisticPointsContainer(UserDict):
+class StatisticPointsContainer(UserDict):  # type: ignore
```

Comment:
Why do we need this?

Reply:
When I tried putting `UserDict[Any, Any]` earlier, I was encountering some error which I am unable to reproduce now, I assume it was regarding some other type error. Hence, I updated it now to `UserDict[Any, Any]`.